src/Generators/Microsoft.Gen.AutoClient/Emitter.cs

```diff
@@ -489,4 +517,39 @@ private void EmitClassVariablesAndConstructor(RestApiType restApiType, string ht
         OutLn($"{optionsName} = autoClientOptions;");
         OutCloseBrace();
     }
+
+    private void EmitPathParameterValidation(string staticsName)
+    {
+        OutLn($$"""
+            private static bool IsPathParameterValid(string value)
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    return false;
+                }
+
+                if (value.IndexOfAny({{staticsName}}.Slashes) > -1)
+                {
+                    return false;
+                }
+            
+                var i = 0;
+
+                // Skip whitespaces
+                while (i < value.Length && char.IsWhiteSpace(value[i])) i++;
+
+                // If it contains a dot it should only be made of dots if not ending with whitespaces
+                while (i < value.Length && value[i] == '.') i++;
+
+                while (i < value.Length && char.IsWhiteSpace(value[i])) i++;
+
+                if (i == value.Length)
+                {
+                    return false;
+                }
+            
+                return true;
```

Comment:
```suggestion
var sp = value.AsSpan().Trim();

return (sp.Length != 0 && sp.IndexOf("\\") == -1 && (sp.IndexOf('.') == -1 || sp.IndexOfAnyExcept('.') == -1))
```
would that work? untested

> // If it contains a dot it should only be made of dots if not ending with whitespaces

I read this as, if it contains a dot the dots must be contiguous and be followed by nothing but possibly whitespace?

Reply:
> I read this as, if it contains a dot the dots must be contiguous and be followed by nothing but possibly whitespace?

Correct

> would that work? untested

There is no such `ROS<string>.Trim()`, only one that takes a `char/char[]` to be trimmed. But that would make the whole thing much simpler since the complexity is to prevent using `string.Trim()`.

@GrabYourPitchforks would that be ok to check specifically for `new char[] { ' ', '\t', '\n', '\r' };` or a subset of this even?

Reply:
There are other whitespace chars beyond what you listed. For simplicity, if you want to go this route, you should just call the `Trim` API. It's available across all runtimes `ROS<char>` is available on: https://apisof.net/catalog/65462c87-fbb3-e410-a5b0-e57bcdae22f1

Reply:
FWIW, since you're writing a generator, you may need to invoke the extension method by its fully qualified name rather than as an instance method.

```cs
System.ReadOnlySpan<char> trimmed = System.MemoryExtensions.Trim(value.AsSpan());
```

Reply:
> There is no such ROS<string>.Trim(),

As Mr G. Y. Pitchforks suggested it’s an expansion method (what I pasted does compile)

Reply:
`IndexOfAnyExcept` is only .NET7+, so substituted it in other cases.